File: java/leetcode/counting-bits.java
class Solution {
    public int[] countBits(int num) {
        int[] result = new int[num + 1];
        for (int i = 0; i <= num; i++) {
            result[i] = countOnes(i);
        }
        return result;
    }

    private int countOnes(int thisNum) {
        int countOne = 0;
        while(thisNum > 0) {
            if (thisNum % 2 != 0) { countOne++; }
            thisNum = Math.round(thisNum / 2);
        }

        return countOne;
    }
}

File: python/leetcode/counting-bits.py
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        result = []
        for i in range(num + 1):
            result.append(self.countOnes(i))
        return result

    def countOnes(self, this_num):
        count_one = 0
        while this_num > 0:
            if (this_num % 2 != 0):
                count_one += 1
            this_num = this_num // 2
        return count_one
